kafka/consume.go

```diff
@@ -73,33 +81,91 @@ func NewConsumer(log logrus.FieldLogger, conf Config, opts ...ConfigOpt) (*Confl
 		return nil, err
 	}
 
-	confluenceConsumer := &ConfluentConsumer{
+	if conf.Timeout == 0 {
+		conf.Timeout = DefaultTimout
+	}
+
+	cc := &ConfluentConsumer{
 		c:    consumer,
 		conf: conf,
 		log:  log,
 	}
 
-	return confluenceConsumer, nil
+	if cc.conf.Consumer.Partition == nil {
+		// if we are not reading from a specific partition, we subscribe
+		cc.setupRebalanceHandler(cc.conf.Consumer.InitialOffset)
+		cc.log.WithField("kafka_topic", cc.conf.Topic).Debug("Subscribing to Kafka topic")
+		cc.rebalanceHandlerMutex.Lock()
+		defer cc.rebalanceHandlerMutex.Unlock()
+		if serr := cc.c.Subscribe(cc.conf.Topic, cc.rebalanceHandler); serr != nil {
+			err = errors.Wrap(serr, "error subscribing to topic")
```

Comment:
Minor but you can do an early return here instead of delegating the error handling